From aeb1a99187c39b2ddfc9fceda325599021fa6a37 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Wed, 5 Apr 2023 12:38:11 +0200
Subject: [PATCH 0393/1306] ddraw: Remove topmost from the previous window when
 releasing exclusive cooperative level.

Fixes The Binding of Isaac staying topmost, regression after upstream
commit e44afcd8bf17eb628fcb1ea0b012970b6e611da4.

Note: this is possibly a hack and maybe we should use exclusive_window
instead, but it is cleared when focus is lost in device_parent_activate,
and removing topmost there as well is a bigger, riskier change.

CW-Bug-Id: #22108
---
 dlls/ddraw/ddraw.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/ddraw/ddraw.c b/dlls/ddraw/ddraw.c
index 5887854556b..8603f4a1113 100644
--- a/dlls/ddraw/ddraw.c
+++ b/dlls/ddraw/ddraw.c
@@ -978,7 +978,7 @@ static HRESULT ddraw_set_cooperative_level(struct ddraw *ddraw, HWND window,
            topmost bit unless the DDSCL_NOWINDOWCHANGES flag is set in
            this call that sets it to normal, not in the old coop level. */
         if (!(cooplevel & DDSCL_NOWINDOWCHANGES))
-            SetWindowPos(window, HWND_NOTOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE);
+            SetWindowPos(ddraw->dest_window, HWND_NOTOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE);
 
         if (restore_mode_on_normal && FAILED(ddraw7_RestoreDisplayMode(&ddraw->IDirectDraw7_iface)))
             ERR("RestoreDisplayMode failed\n");
-- 
2.43.2

